src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -519,21 +565,34 @@ internal Result CreateResult(ResultVersionOne v1Result)
             {
                 result = new Result
                 {
-                    AnalysisTarget = CreateFileLocation(v1Result.Locations?[0]?.AnalysisTarget),
                     BaselineState = Utilities.CreateBaselineState(v1Result.BaselineState),
+                    Fixes = v1Result.Fixes?.Select(CreateFix).ToList(),
                     Id = v1Result.Id,
                     Level = Utilities.CreateResultLevel(v1Result.Level),
                     Locations = v1Result.Locations?.Select(CreateLocation).ToList(),
                     Message = CreateMessage(v1Result.Message),
                     Properties = v1Result.Properties,
                     RuleId = v1Result.RuleId,
-                    RuleMessageId = v1Result.FormattedRuleMessage.FormatId,
                     Stacks = v1Result.Stacks?.Select(CreateStack).ToList(),
                     SuppressionStates = Utilities.CreateSuppressionStates(v1Result.SuppressionStates),
                 };
 
-                // TODO: transfer formatted rule messages
+                if (v1Result.Locations?[0]?.AnalysisTarget?.Uri != v1Result.Locations?[0]?.ResultFile?.Uri)
+                {
+                    result.AnalysisTarget = CreateFileLocation(v1Result.Locations[0].AnalysisTarget);
+                }
```

Comment:
This is worth a comment: "The spec says that analysisTarget is required only if it differs from the result file."

Reply:
OK